### PR TITLE
feat(observation): complete encrypted observation and advice

### LINKED ISSUE #12: Complete encrypted observation, durable verification, and evidence-backed advice in PR #11
## Duplicate search

Searched all repository issues and open pull requests for observation content retention, encrypted workspace binding, verification jobs/check policy, advice status, and migration `0003`. Issues #2, #5, and #6 cover other setup/provider work and are not duplicates. PR #11 is the existing implementation branch this issue governs.

## Problem

PR #11 has established the initial observation seams, logical identity, schema-3 storage, outbox recovery/quarantine, and session lifecycle handling, but it does not yet implement the complete authority contract required for reliable observation and advice.

Yoetz must retain all task-relevant observable context needed for accurate advice while excluding hidden reasoning, platform prompts, credentials, detected secrets, unrelated files, and untethered ambient logs. Retained content must be encrypted immediately, every actual workspace change must receive durable verification coverage, and advice must surface through ordinary findings, status, and safe Codex hooks.

### Authority and threat model

- Amend ADR-009, ADR-010, `specs/INTERFACES.md`, and the owning observation, verification, SQLite, setup, status, privacy, and control-protocol specs.
- Authorize encrypted observation content references, encrypted workspace bindings, exact-digest check-policy trust, verification jobs/results, advice history, and logical identity claims.
- Define authenticated-encryption-at-rest guarantees and explicitly exclude claims of protection against root or a compromised same-user Yoetz/Codex process while the vault is unlocked.
- Keep the six MCP tools unchanged; ingestion, policy trust, verification management, and advice stay on local-control/CLI surfaces.

### Reliable encrypted observation

- Introduce one source-neutral normalized observation record so hook and stream equivalents share semantic kind, correlation identity, roles, operation digest, and logical identity.
- Add bounded encrypted content chunk ingestion and encrypted object manifests, with secret redaction before encryption and no plaintext fallback spool.
- Store normalized workspace locators only as authenticated encrypted objects; keep only commitments and object IDs in plaintext state.
- Make cursor advancement atomic with durable outbox insertion, preserve unknown visible future events as opaque observations plus explicit gaps, allow newer session generations to clear a stopped fence, drain all mapped sessions fairly, and preserve aggregate evidence when quarantine detail is evicted.

### Durable verification and advice

- Add fixed `.yoetz/checks.toml` policy schema `yoetz.approved-check-policy/1`; exact raw digests require trusted-local confirmation and any byte change suspends execution.
- Add `yoetz observe checks preview|trust|status|revoke|run` with exact argv, `shell=False`, sanitized environment, bounded output/time, sandbox enforcement, and fail-closed network behavior.
- Add a generation-fenced verification worker that schedules on material subject-state changes, coalesces rapid changes, persists immutable results, and recovers abandoned jobs after restart.
- Persist redacted check output only in encrypted objects.
- Replace envelope-only advice input with one context builder covering observations/gaps, ledger state, subject-state/changed paths, trusted policy/checks, verification freshness, runtime readiness, and advice history.
- Keep deterministic advice offline; route optional semantic advice only through the existing privacy gateway and effective verification policy.
- Persist advice frontier/history/suppression/delivery/freshness and materialize deterministic advice as ordinary `finding_recorded` events.
- Add ordinary `status` view `advice`, richer trusted-local observe status, and bounded safe hook `additionalContext`.

### Storage and production composition

- Rewrite unmerged migration `0003` to own encrypted workspace/content references, logical identity, trusted policies, verification jobs/leases/results, and advice history/delivery with repository-owned access and required constraints.
- Preserve schema-2 upgrade/readability, restart, backup/restore, and failed-migration rollback behavior.
- Wire real repositories, workspace/subject adapters, check runner, verification worker, context builder, semantic advisor, and coordinator in `ready_composition`; production acceptance tests must exercise the real composition.
- Regenerate generated/frozen artifacts with their owning scripts.

## Security boundary

- No plaintext observation spool.
- No content, absolute paths, secrets, or raw check output in SQLite columns, structural observation JSON, status, logs, diagnostics, backups, hook context, or semantic packets.
- Secret-bearing spans are redacted in memory before object creation; raw values must never reach durable or diagnostic surfaces.
- Encrypted database/object theft without vault keys should reveal only structural metadata and commitments; tampering must fail authentication.
- This does not claim absolute immunity against root access or a compromised same-user process while the vault is unlocked.

## Acceptance criteria

- [ ] Hook/stream equivalence and exactly-once logical materialization are proven.
- [ ] Outbox overflow, unknown events, session-generation recovery, global fair drain, and quarantine aggregate-loss evidence are proven.
- [ ] Hidden reasoning/platform prompts/credentials/unrelated files are rejected; secrets are redacted before encryption.
- [ ] No task plaintext, absolute path, check output, transcript text, or injected secret leaks to plaintext durable/diagnostic surfaces.
- [ ] Stolen files are unreadable without keys and tampering fails authentication; locked-vault capture yields a gap and no plaintext spool.
- [ ] Workspace reopening is descriptor-safe/no-follow and rejects traversal/symlink escape.
- [ ] Untrusted or changed check policy executes nothing; trusted exact digests run exact argv only; network checks fail closed without separate enforced authorization.
- [ ] Every detected material subject-state change receives serialized/coalesced durable verification; restarts recover; stale results never count as current.
- [ ] Deterministic advice works without semantic service; semantic egress is minimized/approved and failures cannot erase deterministic findings.
- [ ] Matching advice finding IDs/evidence frontier appear in ordinary findings, compact, `status view=advice`, local observe status, and safe hooks.
- [ ] Schema-2 upgrade, schema-3 restart, backup/restore, and failed-migration rollback pass.
- [ ] Targeted suites, full pytest, Ruff, pinned Pyright, schema/resource parity, packaging, and installed-wheel production-composition tests pass.
- [ ] Every human and code-review-agent comment on PR #11 is fixed or explicitly dispositioned.
- [ ] PR #11 body documents this issue, authority changes, migration contract, security boundary, verification evidence, and review dispositions.

## Non-goals / accepted boundaries

- No new MCP tool names.
- No plaintext fallback capture queue.
- No direct repository/transcript access for external semantic providers.
- No `cannot be hacked` or unlocked-host compromise resistance claim.
- CI repair and another live `codex-testing` proof are not blockers for this issue; the existing accepted live-proof boundary remains.

## Design gate

This issue changes protocol, privacy/egress, storage/durability, check-execution authority, migration `0003`, ADRs, and production composition. Per `AGENTS.md`, implementation must wait for explicit maintainer acknowledgement on this issue.

**Requested acknowledgement:** approve the scope and authority direction above for continued implementation in existing PR #11 / branch `feat/observation-advice-finish` from reviewed head `4dfaaa1`.

### PR DESCRIPTION
**Pull request by TheGaySupreme123**
## Issue and design gate

- Closes #12.
- Duplicate search is documented in #12.
- Maintainer acknowledgement is recorded in [issue comment 5058047829](https://github.com/TheGaySupreme123/yoetz/issues/12#issuecomment-5058047829) and covers protocol, privacy/egress, storage/durability, migration `0003`, check authority, and production composition.

## Summary

Completes the observation/advice implementation from reviewed head `4dfaaa1`.

- Normalizes equivalent hook and stream observations into one logical identity and stable ledger operation.
- Captures bounded task-visible content, redacts secrets before transport, encrypts content/workspace locators immediately, and records explicit gaps when capture is unavailable.
- Couples stream cursors to durable outbox insertion, preserves unknown events, supports generation-scoped session restart, drains all workspace sessions fairly, and retains aggregate evidence when bounded quarantine detail is evicted.
- Adds exact-byte `.yoetz/checks.toml` policy trust, sandboxed exact-argv checks, durable generation-fenced verification jobs/results, restart recovery, stale-result fencing, and encrypted check output.
- Builds advice from observation/check/composition facts, derives semantic readiness from production provider composition, persists advice history, materializes ordinary findings, and exposes the `advice` status projection plus bounded local/hook guidance.
- Wires the real SQLite repositories, verification worker, coordinator, privacy gateway, and advice builder through ready composition.

## Authority and security boundary

Amends ADR-009, ADR-010, `specs/INTERFACES.md`, and the owning observation, verification, SQLite, setup, status, privacy, control-protocol, packaging, resource, and migration specs.

The supported claim is authenticated encryption at rest, strict local-control boundaries, secret exclusion/redaction before encryption, sandboxed exact-argv verification, and minimized policy-approved semantic egress. This does **not** claim protection against root access or a compromised same-user Yoetz/Codex process while the vault is unlocked. There is no plaintext fallback spool; unavailable capture is structural-only with an explicit gap.

The six MCP tools are unchanged. Observation ingest, policy trust, verification management, and rich diagnostics remain local control/CLI capabilities.

## Migration and generated artifacts

Migration `0003` now owns encrypted workspace/content references, logical-identity claims, trusted policy digests, verification jobs/leases/results, and advice snapshot/history/delivery state with repository-owned access and schema constraints. Schema-2 upgrade and schema-3 restart behavior are covered while older ledger/object data remains readable.

Canonical schemas, packaged migration mirrors, schema/resource manifests, resource-set identity, and runtime-support identity were regenerated/synchronized. The reviewed installed resource inventory is now 73 entries.

## Verification

Passing:

- All current GitHub checks, including `pr-ci-required`, unit/conformance, formatting/type, contracts/resources, package/boundary, SQLite, security, plaintext-canary, and source/artifact boundaries.
- Targeted observation/storage/status/setup/protocol slice: `200 passed`.
- Exact CI unit/property/conformance slice: `1270 passed`.
- Approved-check runner/sandbox unit slice: `7 passed`.
- macOS enforcing-sandbox slice outside nested Codex Seatbelt: `2 passed`.
- Broad suite with the valid managed Python, excluding only the unrelated untouched kill-matrix controller: `1833 passed, 17 skipped, 4 xfailed`.
- Dedicated packaging and installed-wheel suite with the valid managed Python: `193 passed, 3 skipped, 4 xfailed`.
- Version/resource compatibility slice: `22 passed`.
- Ruff format: `402 files already formatted`.
- Ruff lint: `All checks passed!`.
- Pinned Pyright: `0 errors, 0 warnings, 0 informations`.
- Schema parity: `52 schema(s) match`.
- Resource parity: `73 resource(s)`.
- Spec manifest: `628` rows / `614` future owners, pass.

For completeness, a raw local full-suite run under the default shell environment reported `34 failed, 1816 passed, 17 skipped, 4 xfailed`. Nineteen packaging failures came from `uv` selecting a pre-existing incomplete phase-0 managed-Python tree whose standard-library `encodings` sources are absent; pinning uv to the complete managed-Python installation makes the complete packaging suite pass. The other fifteen failures are the untouched kill-matrix helper rejecting Python's `signal.Signals` enum through an exact `type(signal) is int` guard. The controlled PR CI environment is green and does not reproduce those local-environment failures.

## Review disposition

All four review threads were fixed, replied to, and resolved:

1. Stream cursor remains at the last successfully queued envelope on outbox overflow, with replay coverage.
2. Stable operation lookup occurs before payload staging/finalization, preventing orphan encrypted objects on replay.
3. Stream `item.completed` is normalized to hook `PostToolUse` before role and operation-digest selection.
4. `outbox_quarantined` and aggregate eviction evidence are documented in the owning specs.

## Boundaries

- [x] Maintainer acknowledgement recorded on #12.
- [x] Authority/spec changes complete.
- [x] Implementation and generated artifacts complete.
- [x] Required local and CI verification evidence recorded.
- [x] No ignored/private architecture drafting material included.
- [x] Every review comment dispositioned.

### CODE REVIEW

File: src/yoetz/adapters/integrations/codex_session_stream.py [hunk 1]
```diff
@@ -586,19 +580,32 @@ def reconcile_session_stream(
...
+            # Enqueue only after the row is retained; the cursor is persisted
+            # after the whole batch, so it never advances past an unqueued event.
+            if (
+                store.enqueue_outbox(workspace_commitment, codex_session_id, envelope)
+                == ObservationGapCode.OUTBOX_OVERFLOW.value
+            ):
+                overflow = True
         elif result.disposition.value == "duplicate":
             duplicates += 1
     store.set_stream_cursor(workspace_commitment, session_commitment, advance.cursor)
```
[the remark this answers is not in the document]
Comment (TheGaySupreme123) [reply src/yoetz/adapters/integrations/codex_session_stream.py]: Fixed in `011164c`. Stream reconciliation now tracks `committed_cursor` separately and advances it only after `enqueue_outbox()` succeeds. On overflow it leaves the later event replayable and discards the partial tail associated with the uncommitted cursor. The targeted observation suite includes overflow/replay coverage.

File: src/yoetz/application/observation_coordinator.py [hunk 2]
```diff
@@ -161,23 +158,22 @@ async def ingest_request(self, request: ObservationIngestRequest) -> Observation
...
+                # observation row may already exist while a prior ledger append
+                # failed, so re-run the idempotent materialize/append to repair it.
+                # If any step raises, the broad guard below turns it into a
+                # retryable rejection so the outbox keeps the entry pending.
+                batch = materialize_observation_envelope(request.envelope, task_id=runtime.task_id)
                 if batch.skip_reason is None and batch.drafts:
-                    await self._append_materialized(
-                        runtime, request.envelope, batch
-                    )
+                    await self._append_materialized(runtime, request.envelope, batch)
```
[the remark this answers is not in the document]
Comment (TheGaySupreme123) [reply src/yoetz/application/observation_coordinator.py]: Fixed in `011164c`. `_append_materialized()` now derives the stable operation ID and calls `lookup_operation()` before any commitment, staging, or finalization work. Replays return the existing operation identity without creating orphan encrypted payload objects.

File: src/yoetz/application/observation_materialize.py [hunk 3]
```diff
@@ -565,7 +615,7 @@ def observation_operation_digest(
                 "task_id": task_id,
                 "session_id": session_id,
                 "writer_id": writer_id,
-                "source_identity": source_identity,
+                "logical_identity": logical_identity,
                 "roles": draft_roles,
```
[the remark this answers is not in the document]
Comment (TheGaySupreme123) [reply src/yoetz/application/observation_materialize.py:629]: Fixed in `011164c`. A correlated session-stream `item.completed` is normalized to `PostToolUse` before materialization chooses roles, so both sources produce the same `("action", "result")` role tuple, logical identity, operation digest, and stable ledger IDs. Cross-source dedup is covered by coordinator/production-composition tests.

File: src/yoetz/domain/observation.py [hunk 4]
```diff
@@ -179,6 +179,7 @@ class ObservationGapCode(str, Enum):  # noqa: UP042 - exact durable wire enum
     SOURCE_LAG = "source_lag"
     MAPPING_MISSING = "mapping_missing"
     OUTBOX_OVERFLOW = "outbox_overflow"
+    OUTBOX_QUARANTINED = "outbox_quarantined"
```
[the remark this answers is not in the document]
Comment (TheGaySupreme123) [reply src/yoetz/domain/observation.py:197]: Fixed in `011164c`. The owning observation/domain and coordinator specs now define `outbox_quarantined`, permanent-rejection retention, and bounded quarantine eviction with aggregate commitment, count, time range, and explicit loss evidence. The spec manifest gate passes.
